spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java

```diff
@@ -411,7 +411,7 @@ public static void applyQueryOptions(StructuredQuery.Builder builder, DatastoreQ
 		if (queryOptions.getCursor() != null) {
 			builder.setStartCursor(queryOptions.getCursor());
 		}
-		if (queryOptions.getSort() != null && persistentEntity != null) {
+		if (queryOptions.getSort() != null) {
```

Comment:
Do we know that it's not null? I mean, if we get here, it's clearly not null because it would have NPEd on line 392 otherwise. But persistent entity comes from `this.datastoreMappingContext.getPersistentEntity(entityClass)`, and I don't see a null check anywhere.

Reply:
IntelliJ also identified this one and has a helpful "Find Cause" action - and you've nailed it as to why: it's being dereferenced previously in this scope of code.